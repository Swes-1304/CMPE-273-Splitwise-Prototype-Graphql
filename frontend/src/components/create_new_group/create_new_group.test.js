import React from 'react';
import { render, cleanup } from '@testing-library/react';
import TestRenderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import Createnewgroup from './create_new_group';

const component = TestRenderer.create(
  <MemoryRouter>
    <Createnewgroup />
  </MemoryRouter>
);

// eslint-disable-next-line no-undef
afterEach(cleanup);

it('renders', async () => {
  expect(component.toJSON()).toMatchSnapshot();
});

test('Check for create button', async () => {
  const { getByTestId } = render(
    <MemoryRouter>
      <Createnewgroup />
    </MemoryRouter>
  );
  expect(getByTestId('Create')).toHaveTextContent('Save');
});
