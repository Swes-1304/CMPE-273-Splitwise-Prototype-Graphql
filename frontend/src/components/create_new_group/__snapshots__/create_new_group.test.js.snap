// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div>
  <div>
    <nav
      className="navbar-default navbar navbar-expand navbar-light"
    >
      <a
        className="navbar-brand"
        classname="Navbar-Brand"
        href="/"
        variant="light"
      >
        <img
          alt="icon"
          className="icon"
          src="/Group_Photos/Splitwiseicon.png"
        />
        <span
          className="NavItem"
        >
          Splitwise
        </span>
      </a>
      <div
        aria-expanded={null}
        className="justify-content-end navbar-collapse collapse"
      >
        <div
          className="navbar-nav"
          onKeyDown={[Function]}
        >
          <ul
            className="nav navbar-nav navbar-right"
          >
            <li>
              <button
                className="login-default btn btn-primary"
                disabled={false}
                type="button"
              >
                <a
                  href="/login"
                  onClick={[Function]}
                >
                  Login 
                </a>
              </button>
              or
               
              <button
                className="Signup-default btn btn-primary"
                disabled={false}
                type="button"
              >
                <a
                  href="/signup"
                  onClick={[Function]}
                >
                  Sign Up
                </a>
              </button>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  </div>
  <div
    className="profilepage-block"
  >
    <section
      className=" createpage-blocksec"
      style={
        Object {
          "float": "left",
          "height": "300px",
          "padding-left": "20px",
          "padding-top": "1px",
          "width": "300px",
        }
      }
    >
      <div
        id="creategroup_avatar"
      >
        <img
          alt="group pic"
          className="grpavatar"
          src="/Group_photos/default_avatar.png"
        />
        <br />
        <label
          htmlFor="group_avatar"
        >
          Change your group avatar 
          <br />
          <input
            id="group_avatar"
            name="group_avatar"
            onChange={[Function]}
            type="file"
          />
        </label>
        <br />
      </div>
    </section>
    <section
      className="create-group-box"
    >
      <section
        className="right-box"
        style={
          Object {
            "float": "right",
            "padding-left": "20px",
            "padding-top": "1px",
            "width": "350px",
          }
        }
      >
        <form
          className="groupform"
          id="groupform"
        >
          <div
            className="createnewgroup"
          >
            <h2>
              START A NEW GROUP
            </h2>
            <div>
              <h3>
                My group shall be called....
              </h3>
              <input
                id="group_name"
                name="group_name"
                onChange={[Function]}
                required={true}
                type="text"
              />
            </div>
            <br />
            <div
              className="group_members"
            >
              <div
                className="users"
              >
                <h2>
                  Group members
                </h2>
                <div
                  className="group-member"
                >
                  <div
                    className="grpnameemail"
                  >
                    (
                    <em />
                    )
                  </div>
                  <div
                    className="grpnameemail"
                    style={
                      Object {
                        "display": "flex",
                        "flex-direction": "row",
                        "width": "300px",
                      }
                    }
                  >
                    <div
                      className="grpnameemail"
                      style={
                        Object {
                          "width": "300px",
                        }
                      }
                    >
                      <div
                        className="div-select css-2b097c-container"
                        id="group_members_1_username"
                        onKeyDown={[Function]}
                      >
                        <span
                          aria-atomic="false"
                          aria-live="polite"
                          aria-relevant="additions text"
                          className="css-1f43avz-a11yText-A11yText"
                        />
                        <div
                          className=" css-yk16xz-control"
                          onMouseDown={[Function]}
                          onTouchEnd={[Function]}
                        >
                          <div
                            className=" css-g1d714-ValueContainer"
                          >
                            <div
                              className=" css-1wa3eu0-placeholder"
                            >
                              Select...
                            </div>
                            <div
                              className="css-b8ldur-Input"
                            >
                              <div
                                className=""
                                style={
                                  Object {
                                    "display": "inline-block",
                                  }
                                }
                              >
                                <input
                                  aria-autocomplete="list"
                                  autoCapitalize="none"
                                  autoComplete="off"
                                  autoCorrect="off"
                                  disabled={false}
                                  id="react-select-2-input"
                                  onBlur={[Function]}
                                  onChange={[Function]}
                                  onFocus={[Function]}
                                  spellCheck="false"
                                  style={
                                    Object {
                                      "background": 0,
                                      "border": 0,
                                      "boxSizing": "content-box",
                                      "color": "inherit",
                                      "fontSize": "inherit",
                                      "label": "input",
                                      "opacity": 1,
                                      "outline": 0,
                                      "padding": 0,
                                      "width": "1px",
                                    }
                                  }
                                  tabIndex="0"
                                  type="text"
                                  value=""
                                />
                                <div
                                  style={
                                    Object {
                                      "height": 0,
                                      "left": 0,
                                      "overflow": "scroll",
                                      "position": "absolute",
                                      "top": 0,
                                      "visibility": "hidden",
                                      "whiteSpace": "pre",
                                    }
                                  }
                                >
                                  
                                </div>
                              </div>
                            </div>
                          </div>
                          <div
                            className=" css-1hb7zxy-IndicatorsContainer"
                          >
                            <span
                              className=" css-1okebmr-indicatorSeparator"
                            />
                            <div
                              aria-hidden="true"
                              className=" css-tlfecz-indicatorContainer"
                              onMouseDown={[Function]}
                              onTouchEnd={[Function]}
                            >
                              <svg
                                aria-hidden="true"
                                className="css-tj5bde-Svg"
                                focusable="false"
                                height={20}
                                viewBox="0 0 20 20"
                                width={20}
                              >
                                <path
                                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                                />
                              </svg>
                            </div>
                          </div>
                        </div>
                        <input
                          name="group_members_1_username"
                          type="hidden"
                          value=""
                        />
                      </div>
                    </div>
                    <button
                      className="removegm"
                      name="removegm"
                      onClick={[Function]}
                      style={
                        Object {
                          "background-color": "white",
                          "border": "none",
                          "color": "#ff652f",
                          "font-weight": "bolder",
                        }
                      }
                      type="button"
                    >
                      X
                    </button>
                  </div>
                  <button
                    className="addgm"
                    onClick={[Function]}
                    style={
                      Object {
                        "background-color": "white",
                        "border": "none",
                        "color": "#3ac3ab",
                        "font-weight": "bolder",
                      }
                    }
                    type="button"
                  >
                    + Add Person
                  </button>
                </div>
              </div>
              <div
                id="invite_link_container"
              >
                <div
                  className="invitelink"
                >
                  <div />
                </div>
                <div
                  className="savebtn"
                >
                  <button
                    className="Signup-default"
                    data-testid="Create"
                    formNoValidate={true}
                    onClick={[Function]}
                    type="button"
                  >
                    Save
                  </button>
                  <p
                    className="errmsg"
                    style={
                      Object {
                        "color": "maroon",
                      }
                    }
                  >
                     
                     
                  </p>
                </div>
              </div>
            </div>
          </div>
        </form>
      </section>
    </section>
  </div>
</div>
`;
